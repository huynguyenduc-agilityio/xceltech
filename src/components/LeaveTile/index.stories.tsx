import type { Meta, StoryObj } from '@storybook/react';

// Components
import LeaveTile from '.';

const meta = {
  title: 'Components/LeaveTile',
  component: LeaveTile,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <LeaveTile {...args} />
    </div>
  ),
} satisfies Meta<typeof LeaveTile>;

export default meta;

type Story = StoryObj<typeof LeaveTile>;

export const Default: Story = {
  args: {
    title: 'Annual Leave',
    count: 60,
  },
};

export const WithThousands: Story = {
  args: {
    title: 'Annual Leave',
    count: 60000,
  },
};

export const WithMillions: Story = {
  args: {
    title: 'Annual Leave',
    count: 6000000,
  },
};

export const WithBillions: Story = {
  args: {
    title: 'Annual Leave',
    count: 6000000000,
  },
};
