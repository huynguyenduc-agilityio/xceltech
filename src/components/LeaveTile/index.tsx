import { cn, formatNumberWithMetricPrefix } from '@/utils';

export interface LeaveTileProps {
  count: number;
  title: string;
}

const LeaveTile = ({ count, title }: LeaveTileProps) => {
  const formattedCount = formatNumberWithMetricPrefix(count);
  let fontSize = 'text-[70px]';

  if (formattedCount.length > 3) {
    fontSize = 'text-[50px]';
  } else if (formattedCount.length > 2) {
    fontSize = 'text-[60px]';
  }

  return (
    <div
      className={
        'flex items-center gap-12 w-[445px] h-[182px] px-5 py-6 rounded-[16px] bg-primary'
      }
    >
      <div className="flex items-center justify-center w-[135px] h-[135px] rounded-full bg-white">
        <span className={cn('text-primary', fontSize)}>{formattedCount}</span>
      </div>
      <div className="flex flex-col items-center">
        <span className="text-xl font-bold text-white">{title}</span>
        {/* TODO: Button  */}
      </div>
    </div>
  );
};

export default LeaveTile;
