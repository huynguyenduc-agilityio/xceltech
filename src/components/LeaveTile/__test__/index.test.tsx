import { render } from '@testing-library/react';

// Components
import LeaveTile from '..';

describe('LeaveTile component', () => {
  it('renders the Button with children', () => {
    const { getByText } = render(<LeaveTile title="Annual Leave" count={60} />);

    expect(getByText('Annual Leave')).toBeInTheDocument();
    expect(getByText('60')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <LeaveTile title="Annual Leave" count={60} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
