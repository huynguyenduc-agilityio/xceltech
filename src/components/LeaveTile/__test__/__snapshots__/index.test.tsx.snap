// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveTile component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-12 w-[445px] h-[182px] px-5 py-6 rounded-[16px] bg-primary"
  >
    <div
      class="flex items-center justify-center w-[135px] h-[135px] rounded-full bg-white"
    >
      <span
        class="text-primary text-[70px]"
      >
        60
      </span>
    </div>
    <div
      class="flex flex-col items-center"
    >
      <span
        class="text-xl font-bold text-white"
      >
        Annual Leave
      </span>
    </div>
  </div>
</DocumentFragment>
`;
